src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -53,38 +60,51 @@
 
     private final AuthorizationCheckService authCheckService;
 
+    private final FileUploadSubmissionRepository fileUploadSubmissionRepository;
+
     private final TokenProvider tokenProvider;
 
+    private final FileUploadExerciseRepository fileUploadExerciseRepository;
+
     public FileResource(FileService fileService, ResourceLoader resourceLoader, UserService userService, AuthorizationCheckService authCheckService,
-            LectureRepository lectureRepository, TokenProvider tokenProvider) {
+            LectureRepository lectureRepository, TokenProvider tokenProvider, FileUploadSubmissionRepository fileUploadSubmissionRepository,
+            FileUploadExerciseRepository fileUploadExerciseRepository) {
         this.fileService = fileService;
         this.resourceLoader = resourceLoader;
         this.userService = userService;
         this.authCheckService = authCheckService;
         this.lectureRepository = lectureRepository;
         this.tokenProvider = tokenProvider;
+        this.fileUploadSubmissionRepository = fileUploadSubmissionRepository;
+        this.fileUploadExerciseRepository = fileUploadExerciseRepository;
     }
 
     /**
      * POST /fileUpload : Upload a new file.
      *
      * @param file The file to save
      * @param keepFileName specifies if original file name should be kept
+     * @param isSubmission specifies if file is submitted by a student
      * @return The path of the file
      * @throws URISyntaxException if response path can't be converted into URI
      */
     @PostMapping("/fileUpload")
-    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA')")
-    public ResponseEntity<String> saveFile(@RequestParam(value = "file") MultipartFile file, @RequestParam("keepFileName") Boolean keepFileName) throws URISyntaxException {
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA', 'USER')")
```

Comment:
This creates a problem, because now any user can upload a file. I think we need to add additional checks now, otherwise this endpoint becomes unsecure!

Reply:
Actually, I think we nee a dedicated endpoint for uploading and creating temporary files for the upload exercise. Students could bombard the server with files for the same exercise by manually calling the endpoint. 
They could even just overwrite other submission in theory if they are using filenames chosen by other users. The temporary file would overwrite the actual submission before the other user calls the saveExercise endpoint

Reply:
This sounds much better in terms of security. However we should try to realize this with the most pragmatic approach. It might e.g. be difficult to block a user. But I’m curious how you would implement this 😉

Reply:
So right now, there is a check for the following properties:

- File size
- Is the course the same as for the exerciseId?
- Check whtether the user already uploaded

As long as those 3 don't pass, nothing will be saved.

Reply:
please remove 'USER' in the pre authorize. I guess we could even remove 'TA' here, but I am not sure. Please double check if 'TA' is necessary

Reply:
This endpoint is used for lecture attachments, uploading course icon, quiz exercise generation, editing questions for quiz exercise. At least the last one is accessible for tutor from the client